pkg/service/backup/restore_worker_tables.go

```diff
@@ -453,6 +455,26 @@ func (w *tablesWorker) newRunProgress(ctx context.Context, run *RestoreRun, targ
 	return pr, nil
 }
 
+func (w *tablesWorker) countBatchSize(ctx context.Context, batch []string, host, srcDir string) int64 {
+	var batchSize int64
+	for _, file := range batch {
+		fi, err := w.Client.RcloneFileInfo(ctx, host, srcDir+"/"+file)
+		if err != nil {
+			w.Logger.Error(ctx, "Failed to check file size", "err", err)
+		}
+		batchSize += fi.Size
+	}
+	return batchSize
+}
```

Comment:
It's actually valid comment.
Current approach requires many calls to the cloud storage. It can be rate-limited on the cloud provider side at some moment.
I will replace it with just a single call.

Reply:
I'll make a fix after merging this one https://github.com/scylladb/scylla-manager/pull/3353

Reply:
done, filesizes are cached now